src/hocon_schema.erl

```diff
@@ -505,10 +509,24 @@ map_fields([{FieldName, FieldSchema} | Fields], Conf0, Acc, Opts) ->
 map_one_field(FieldType, FieldSchema, FieldValue, Opts) ->
     Converter = field_schema(FieldSchema, converter),
     {Acc, NewValue} = map_field_maybe_convert(FieldType, FieldSchema, FieldValue, Opts, Converter),
+    NoConversion = maps:get(no_conversion, Opts, false),
+    Validators =
+        case is_primitive_type(FieldType) of
+            true ->
+                %% primitive values are already validated
+                [];
+            false ->
+                %% otherwise valdiate using the schema defined callbacks
+                validators(field_schema(FieldSchema, validator))
+        end,
     case find_errors(Acc) of
+        ok when NoConversion ->
+            %% when no_conversion, we are only filling default values (recursively)
```

Comment:
I don't have much context here, 
`no_conversion` means use default values? Shall we change it to `use_default_val`?

Reply:
changed the option to `only_fill_defaults`